src/main/java/org/ambraproject/wombat/service/remote/SolrQueryBuilder.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2017 Public Library of Science
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package org.ambraproject.wombat.service.remote;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import com.google.common.base.Joiner;
+import com.google.common.base.Strings;
+import org.ambraproject.wombat.util.UrlParamBuilder;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.http.NameValuePair;
+
+public class SolrQueryBuilder {
+  public static List<NameValuePair> buildParameters(ArticleSearchQuery asq) {
+    UrlParamBuilder params = UrlParamBuilder.params();
+
+    params.add("wt", "json");
+
+    if (asq.isPartialSearch()) {
+      params.add("qf", "doc_partial_body");
+      params.add("fl", "*");
+      params.add("fq", "doc_type:partial");
+    } else {
+      params.add("fq", "doc_type:full");
+    }
+
+    params.add("fq", "!article_type_facet:\"Issue Image\"");
+
+    if (asq.getStart() > 0) {
+      params.add("start", Integer.toString(asq.getStart()));
+    }
+    params.add("rows", Integer.toString(asq.getRows()));
+
+    params.add("hl", "false");
+
+    String queryString = asq.getQuery();
+    params.add("q", queryString);
+    if (asq.isSimple()) {
+      // Use the dismax query parser, recommended for all user-entered queries.
+      // See https://wiki.apache.org/solr/DisMax
+      params.add("defType", "dismax");
+    }
+
+    if (asq.getFacetFields().size() > 0) {
+      params.add("facet", "true");
+      for (String field: asq.getFacetFields()) {
+        params.add("facet.field", field);
+      }
+      params.add("facet.mincount", Integer.toString(asq.getFacetMinCount()));
+      params.add("facet.limit", Integer.toString(asq.getFacetLimit()));
+      /* Use a thread per facet. */
+      params.add("facet.threads", Integer.toString(-1));
```

Comment:
Thanks for explaining the magic number! :thread: :thread: :thread:

Reply:
Makes me think I should use a constant....

Reply:
12d2d50e2